src/libs/antares/study/area/area.h

```diff
@@ -60,6 +60,7 @@ class Area final : private Yuni::NonCopyable<Area>
     using Vector = std::vector<Area*>;
     using VectorConst = std::vector<const Area*>;
     using List = std::list<Area*>;
+    using ScratchMap = std::map<const Area*, AreaScratchpad*>;
```

Comment:
Using a raw pointer as a map key works, but it seems a bit clumsy. Indeed, `std::map` is sorted. Depending on (arbitrary) pointer values, elements could be sorted differently.

As for the values, why do you use a raw pointer ?

Reply:
Fixed, now using a reference for AreaScratchpad